lxd/api_metrics.go

```diff
@@ -286,6 +286,27 @@ func metricsGet(d *Daemon, r *http.Request) response.Response {
 
 	metricsCacheLock.Unlock()
 
+	return getFilteredMetrics(s, r, compress, metricSet)
+}
+
+func getFilteredMetrics(s *state.State, r *http.Request, compress bool, metricSet *metrics.MetricSet) response.Response {
+	if !s.GlobalConfig.MetricsAuthentication() {
+		return response.SyncResponsePlain(true, compress, metricSet.String())
+	}
+
+	// Get instances the user is allowed to view.
+	userHasPermission, err := s.Authorizer.GetPermissionChecker(r.Context(), r, auth.EntitlementCanView, auth.ObjectTypeInstance)
+	if err != nil && !api.StatusErrorCheck(err, http.StatusForbidden) {
+		return response.SmartError(err)
+	} else if err != nil {
+		// This is counterintuitive. We are unauthorized to get a permission checker for viewing instances because a metric type certificate
+		// can't view instances. However, in order to get to this point we must already have auth.EntitlementCanViewMetrics. So we can view
```

Comment:
So if you can view all metrics with `auth.EntitlementCanViewMetrics` why cant you also do filtering?